htdocs/categories/class/categorie.class.php

```diff
@@ -1522,7 +1522,7 @@ function getNomUrl($withpicto=0,$option='',$maxlength=0)
 			if (colorIsLight($this->color)) $forced_color='categtextblack';
 		}
 
-        $link = '<a href="'.DOL_URL_ROOT.'/categories/viewcat.php?id='.$this->id.'&type='.$this->type.'" title="'.dol_escape_htmltag($label, 1).'" class="classfortooltip '.$forced_color .'">';
+        $link = '<a href="'.DOL_URL_ROOT.'/categories/viewcat.php?id='.$this->id.'&type='.$this->type.'&linkback='.$_SERVER['PHP_SELF'].'" title="'.dol_escape_htmltag($label, 1).'" class="classfortooltip '.$forced_color .'">';
```

Comment:
The common parameter to go back to an url is 
backtopage=
also you must use urlencode around the url